backend/model/model.go

```diff
@@ -1522,6 +1523,37 @@ func MigrateDB(ctx context.Context, DB *gorm.DB) (bool, error) {
 		return false, e.Wrap(err, "Error setting session id sequence to 30000000")
 	}
 
+	if err := DB.Exec(`
+		CREATE TABLE IF NOT EXISTS error_object_embeddings_partitioned
+		(LIKE error_object_embeddings INCLUDING DEFAULTS INCLUDING IDENTITY)
+		PARTITION BY RANGE (project_id);
+	`).Error; err != nil {
+		return false, e.Wrap(err, "Error creating error_object_embeddings_partitioned")
+	}
+
+	var lastVal int
+	if err := DB.Raw("SELECT coalesce(max(id), 1) FROM projects").Scan(&lastVal).Error; err != nil {
+		return false, e.Wrap(err, "Error selecting max project id")
+	}
+	partitionSize := 1
+	start := lastVal / partitionSize * partitionSize
+
+	// Make sure partitions are created for the next 1k projects
+	for i := 0; i < 1000; i++ {
+		end := start + partitionSize
+		sql := fmt.Sprintf(`
+			CREATE TABLE IF NOT EXISTS error_object_embeddings_partitioned_%d
+			PARTITION OF error_object_embeddings_partitioned
+			FOR VALUES FROM (%d) TO (%d);
```

Comment:
if using `PARTITION BY LIST`:
https://www.postgresql.org/docs/current/sql-createtable.html#:~:text=)%20FOR%20VALUES%20IN%20(%27a%27%2C%20%27b%27)%3B

Reply:
updated to `FOR VALUES IN ('%d');`